packages/utilities/fast-design/src/color/swatch.ts

```diff
@@ -0,0 +1,150 @@
+import { ColorRGBA64, rgbToRelativeLuminance } from "@microsoft/fast-colors";
+import { contrast, RelativeLuminance } from "./utilities/relative-luminance.js";
+
+/**
+ * Represents a color in a {@link Palette}.
+ *
+ * @public
+ */
+export interface Swatch extends RelativeLuminance {
+    /**
+     * Gets a string representation of the color.
+     */
+    toColorString(): string;
+
+    /**
+     * Gets the contrast between this Swatch and another one.
+     * @param target - The relative luminance for comparison
+     */
+    contrast(target: RelativeLuminance): number;
+}
+
+/**
+ * Represents a color in a {@link Palette} using RGB.
+ *
+ * @public
+ */
+export class SwatchRGB implements Swatch {
```

Comment:
Primarily curious around the reasoning behind the change here and what we get for it and what we give up. That give and take is especially interesting as we consider any potential performance implications.

Reply:
The current Swatch/Palette implementation has been limiting, even when building the updated algorithm for Fluent UI. The issue there was I needed one palette that had all of the colors representing the curve, then another built as a subset of the first. I could only do that because the code was internal to the `Impl` class and I could access the constructor.

Much of the `PaletteRGBImpl` class was of common need. That is, everything about the Palette interface. The fact that it's built on Swatches and that you can call some functions to get colors in certain ways. That part does not change based on the actual colors going into the palette. With that understanding. the subclass palette logic should be about getting colors into it. The `BasePalette` should apply to any type of palette I can foresee, and if someone does need to do one thing differently then can override the necessary function.

Swatch is the same way. For instance, when I started working on the opacity prototype I had to copy the entire Swatch impl just to make a couple tiny changes. I think some base functionality of a Swatch should exist that consumers can use and extend directly.

Lastly, a more generic consideration here is that when we built the first version of this we wanted to allow for something other than RGB. In the realm of web-tech I'm not sure that will ever be useful. All displays are RGB-based, and they probably will be for a long time, even for HDR. It's possible to use another color model to generate the palette today, as we do with CIELAB now. It needs to be converted back to RGB though anyway. So part of the thought here is moving the base definition up to the "RGB" level as the most useful starting point for extension. I left the underlying interfaces "in case".

I'm curious about performance implications. Something about classes and interfaces or the ability to tree-shake one implementation over the other?